specification/network/resource-manager/Microsoft.Network/stable/2024-07-01/networkVirtualAppliance.json

```diff
@@ -360,6 +360,93 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkVirtualAppliances/{networkVirtualApplianceName}/getBootDiagnosticLogs": {
+      "post": {
+        "tags": [
+          "NetworkVirtualAppliances"
+        ],
+        "operationId": "NetworkVirtualAppliances_GetBootDiagnosticLogs",
+        "description": "Gets the boot diagnostic logs for a VM instance belonging to the specified Network Virtual Appliance.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "networkVirtualApplianceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of Network Virtual Appliance."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "instanceId",
+            "in": "body",
+            "required":true,
+            "schema": {
+              "$ref": "#/definitions/NetworkVirtualApplianceInstanceId"
+            },
+            "description": "Specifies a virtual machine instance ID from the Network Virtual Appliance VM instances."
+          },
+          {
+            "name": "serialConsoleStorageSasUrl",
+            "in": "body",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/SerialConsoleStorageSasUrl"
+            },
+            "description": "Specifies the sas-url to the storage blob into which serial console logs for the requested instance will be written"
+          },
+          {
+            "name": "consoleScreenshotStorageSasUrl",
+            "in": "body",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/ConsoleScreenshotStorageSasUrl"
+            },
```

Comment:
>202 [](http://example.com/codeflow?start=11&length=3)

Correct async response statis with response body is 200.

Reply:
We return '202 Accepted' because in most cases the request is processed asynchronously and may not have completed yet. A "200 OK" indicates a successful completion of the request. But both would have a response body in this case indicating the instance id on which request succeeded or was accepted. I have added "200 OK" as well.

Reply:
AN API must either be sync or async and not both. So please plan to actually return only a 202 in this case. The 200 status code must still be documented to indicate to the client what the final schema of the response will be to allow it to deserialize the object. So having 200 and 202 documented here is the right thing to do for now.